src/Limiters/quasimonotone.jl

```diff
@@ -198,34 +228,20 @@ function compute_bounds!(
     limiter::QuasiMonotoneLimiter,
     ρq::Fields.Field,
     ρ::Fields.Field,
-)
-    topology = Spaces.topology(axes(ρq))
-    compute_bounds!(
-        limiter,
-        Fields.field_values(ρq),
-        Fields.field_values(ρ),
-        topology,
-    )
-end
-function compute_bounds!(
-    limiter::QuasiMonotoneLimiter,
-    ρq,
-    ρ,
-    topology::Topologies.AbstractTopology,
 )
     compute_element_bounds!(limiter, ρq, ρ)
     if limiter.ghost_buffer isa Spaces.GhostBuffer
         Spaces.fill_send_buffer!(
-            topology,
+            Spaces.topology(axes(ρq)),
             limiter.q_bounds,
             limiter.ghost_buffer,
         )
         ClimaComms.start(limiter.ghost_buffer.graph_context)
     end
-    compute_neighbor_bounds_local!(limiter, topology)
+    compute_neighbor_bounds_local!(limiter, ρ)
     if limiter.ghost_buffer isa Spaces.GhostBuffer
         ClimaComms.finish(ghost_buffer.graph_context)
-        compute_neighbor_bounds_ghost!(limiter, topology)
+        compute_neighbor_bounds_ghost!(limiter, Spaces.topology(axes(ρq)))
     end
 end
```

Comment:
Maybe we can split this function into two methods and dispatch over `limiter.ghost_buffer`

Reply:
I had thought about that, but I didn't change that because of https://github.com/CliMA/ClimaCore.jl/issues/1511, and I spoke with @simonbyrne about the possibility of just removing all things communication in the limiters code, because it's called after `dss!` has been called, so no communication think any communication is needed. Unless we need the entire element from neighboring processes, maybe that is what @simonbyrne was getting at.

I'm happy to change it, though.

Reply:
Thanks. Sounds good.

(I'll try to look at the rest of the PR later)

Reply:
The split method looks like this:
```julia
compute_bounds!(
    limiter::QuasiMonotoneLimiter,
    ρq::Fields.Field,
    ρ::Fields.Field,
) = compute_bounds!(limiter, ρq, ρ, limiter.ghost_buffer)

function compute_bounds!(
    limiter::QuasiMonotoneLimiter,
    ρq::Fields.Field,
    ρ::Fields.Field,
    ghost_buffer::Spaces.GhostBuffer
)
    compute_element_bounds!(limiter, ρq, ρ)
    Spaces.fill_send_buffer!(
        Spaces.topology(axes(ρq)),
        limiter.q_bounds,
        ghost_buffer,
    )
    ClimaComms.start(ghost_buffer.graph_context)
    compute_neighbor_bounds_local!(limiter, ρ)
    ClimaComms.finish(ghost_buffer.graph_context)
    compute_neighbor_bounds_ghost!(limiter, Spaces.topology(axes(ρq)))
end

function compute_bounds!(
    limiter::QuasiMonotoneLimiter,
    ρq::Fields.Field,
    ρ::Fields.Field,
    ::Nothing
)
    compute_element_bounds!(limiter, ρq, ρ)
    compute_neighbor_bounds_local!(limiter, ρ)
end
```
I actually think I prefer the current form with union splitting. idk, what do you think?

Reply:
I don't have strong opinions, especially if the code is likely to change. Pick the one your prefer.